File: client/scalaz/src/main/scala/org/openapitools/client/api/SdtmDatasetVariable.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SdtmDatasetVariable._

case class SdtmDatasetVariable (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
description: Option[String],
role: Option[String],
roleDescription: Option[String],
simpleDatatype: Option[String],
describedValueDomain: Option[String],
links: Option[SdtmDatasetVariableLinks])

object SdtmDatasetVariable {
  import DateTimeCodecs._

  implicit val SdtmDatasetVariableCodecJson: CodecJson[SdtmDatasetVariable] = CodecJson.derive[SdtmDatasetVariable]
  implicit val SdtmDatasetVariableDecoder: EntityDecoder[SdtmDatasetVariable] = jsonOf[SdtmDatasetVariable]
  implicit val SdtmDatasetVariableEncoder: EntityEncoder[SdtmDatasetVariable] = jsonEncoderOf[SdtmDatasetVariable]
}

File: src/cdisc_library_api/specs/sdtm_dataset_variable.clj
(ns cdisc-library-api.specs.sdtm-dataset-variable
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.sdtm-dataset-variable-links :refer :all]
            )
  (:import (java.io File)))


(def sdtm-dataset-variable-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :role) string?
   (ds/opt :roleDescription) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :describedValueDomain) string?
   (ds/opt :_links) sdtm-dataset-variable-links-spec
   })

(def sdtm-dataset-variable-spec
  (ds/spec
    {:name ::sdtm-dataset-variable
     :spec sdtm-dataset-variable-data}))
